python_modules/dagster/dagster/_core/executor/step_delegating/step_delegating_executor.py

```diff
@@ -312,17 +312,18 @@ def execute(self, plan_context: PlanOrchestrationContext, execution_plan: Execut
                                         )
                                     )
                                     if not health_check_result.is_healthy:
-                                        DagsterEvent.step_failure_event(
-                                            step_context=step_context,
-                                            step_failure_data=StepFailureData(
-                                                error=None,
-                                                user_failure_data=None,
-                                            ),
-                                            message=(
-                                                f"Step {step.key} failed health check:"
-                                                f" {health_check_result.unhealthy_reason}"
-                                            ),
+                                        health_check_error = SerializableErrorInfo(
+                                            message=f"Step {step.key} failed health check: {health_check_result.unhealthy_reason}",
+                                            stack=[],
+                                            cls_name=None,
```

Comment:
any concern over health check failing but step still going so now we are running a second copy flavor of problems?

Reply:
maybe some concern, but the checks are usually as simple as 'does entity exist' which should be bread and butter for the docker or k8s api - previously we would have been failing the run in that case which is also not great